### ISSUE TITLE
Wildcard sync using CLI

### ISSUE THREAD

**Issue by muenchdo** [argoproj/argo-cd#2228]
**Is your feature request related to a problem? Please describe.**
We're not quite there yet to have auto-sync enabled for all our applications, so sometimes we call `argocd app sync foo` manually.
We use a single ArgoCD instance to manage multiple clusters, and we prefix the applications using the cluster name, e.g. `eu-central-foo`, `us-west-foo`, etc.
When we update some config across many clusters we need to run the sync command once per cluster.

**Describe the solution you'd like**
It would be awesome if you could run something like `argocd app sync *-foo` which will sync all applications that match the expression `*-foo`.
Maybe this could even be extended to other CLI commands, e.g. `argocd app delete`.

**Have you thought about contributing yourself?**
Yes, I will need some initial guidance though :)

**Comment by alexec** [comment]
Could you write a script that lists the appropriate resources and then iterates over them, e.g.

```bash
for name in $(kubctl -n argocd get app -l my-label=my-value -o name); do 
   argocd app sync $name
done
```

**Comment by muenchdo** [comment]
I think we can consider this closed by #2463.
